### ISSUE TITLE
Images with filenames starting with an accented characters not rendered

### ISSUE THREAD

**Issue by enderlen** [getgrav/grav#2687]
I’ve got an image starting with an accented characters “électricité.png”
Markdown when inserting image is (%C3%A9lectricit%C3%A9.png)
Result grav html is <img src="…/lectricité.png" so no image display because of missing first filename character …

I’m using last version of Grav with all components updated.

Please fix this bug…

**Comment by rhukster** [comment]
The unicode is normal:

<img width="536" alt="monosnap-2019-10-10-13-35-59 image" src="https://user-images.githubusercontent.com/1084697/66600525-f0283500-eb62-11e9-94af-ac09b31e4018.png">

When I drag the image into the editor it outputs: `![](e%CC%81lectricite%CC%81.png)` as the Markdown tag.. This does render correctly though:

<img width="794" alt="monosnap-2019-10-10-13-37-03 image" src="https://user-images.githubusercontent.com/1084697/66600623-1948c580-eb63-11e9-8d89-baa9122a5be2.png">

**Comment by enderlen** [comment]
My result is very different, here is what I get:
![grav_issue_électricité](https://user-images.githubusercontent.com/32302759/66601083-5767d880-eba7-11e9-8cb3-8d4385f82b5a.png)
![grav_issue_électyricité2](https://user-images.githubusercontent.com/32302759/66601280-d6f5a780-eba7-11e9-9d2e-60543e72f5fb.png)

**Comment by rhukster** [comment]
the filename is getting jumbled.  Can you attach a zip of that page (include .md file and all images) so I an try to recreate?

Also what browser are you using?

**Comment by enderlen** [comment]
Here is zip of md+images
[page.zip](https://github.com/getgrav/grav/files/3716262/page.zip)

**Comment by enderlen** [comment]
As a browser I'm using Chrome in French language

**Comment by smz** [comment]
Sorry for dropping in, but I once faced a similar issue (_in a different context_), and I think it might all boils down to you two using different Unicode conventions.

In Unicode some "characters" can be represented in two different forms: a **combined form**, where an "e with acute accent" is represented by a single code point, or a **decomposed form**, where the "_same_" character is represented by a combination of two (or more) code points, in this case an "e" character followed by an acute accent.

Normally characters are "normalized" by the OS to one of the possible forms: Windows (_and I think all *nix OS_) uses what Unicode calls "Normalization Form C", where characters are translated to their decomposed form and finally to their composed form (_single code point_) form, while Mac OS (_and only it, AFAIK..._) uses "Normalization Form D", where characters are always decomposed (_and thus represented by a sequence of code points_).

In my case I had a link pointing to an image (_a similar situation, although I wasn't using Grav at the time_), but the link was written on Windows and thus it was in its NFC form, while the image file came from a Mac and its name (_at the file system level_) was in NFD form, the result being no image being displayed.

From what I see I have the feeling that @rhukster is using a Mac and @enderlen is using Windows or some form of Linux...

Here I think there is something more to check, as it **seems** markdown is "eating" the first composite character.

For more (_and much better!_) information see:
https://en.wikipedia.org/wiki/Precomposed_character
https://en.wikipedia.org/wiki/Unicode_equivalence#Normalization
http://unicode.org/faq/char_combmark.html
https://unicode.org/reports/tr15/#Canon_Compat_Equivalence

Cheers!

**Comment by enderlen** [comment]
Yes I'm on Windows

**Comment by enderlen** [comment]
> the filename is getting jumbled. Can you attach a zip of that page (include .md file and all images) so I an try to recreate?
> 
> Also what browser are you using?

any news? did you succeed to reproduce?

**Comment by smz** [comment]
@rhukster have you tried to replicate this under the original conditions (_i.e. under Windows, not Mac_)?

**Comment by rhukster** [comment]
No I’m afraid I’m on Mac.

**Comment by smz** [comment]
Thanks @rhukster 

This (_you confirming using a Mac_) reinforce my impression that Grav isn't playing well with Unicode Normalization form C...

Please notice how in @enderlen example the "compound" `è` Unicode character seems to be stripped giving `img src="…/lectricité.png"` as the result, while on your Mac the result is correct.

My rough idea is that somewhere a regex *might* trim the string stripping Unicode Form C characters from it, but where that could happen is definitely "_above my pay grade_"! :flushed:

**Comment by smz** [comment]
Hello,

I just stumbled upon this, which I think might be relevant: [The Normalizer class](https://www.php.net/manual/en/class.normalizer.php)

**Comment by mahagr** [comment]
I agree that this must be related to the unicode normalization: if the filename is different from what PHP looks for, the file cannot be found. This is why I never use special characters in filenames myself -- it is just not worth it.

It doesn't really matter if we have normalizer, another filesystem may still encode the name by using different normalization.

**Comment by tg-x** [comment]
Ran into a similar issue with simple text links.

The following link:

```
[Ἀταλάντη καὶ Ἱππομένης.pdf](%E1%BC%88%CF%84%CE%B1%CE%BB%E1%BD%B1%CE%BD%CF%84%CE%B7%20%CE%BA%CE%B1%E1%BD%B6%20%E1%BC%B9%CF%80%CF%80%CE%BF%CE%BC%E1%BD%B3%CE%BD%CE%B7%CF%82.pdf)
```

is rendered as the following:

```
<a href="/some/page/ &#x3BA;&#x3B1;&#x1F76; &#x1F39;&#x3C0;&#x3C0;&#x3BF;&#x3BC;&#x1F73;&#x3BD;&#x3B7;&#x3C2;.pdf">Ἀταλάντη καὶ Ἱππομένης.pdf</a>
```

There's some additional unnecessary escaping that breaks the link.
